modules/ssl/ssl_engine_kernel.c

```diff
@@ -2485,26 +2485,72 @@ static void copy_clienthello_vars(conn_rec *c, SSL *ssl)
 
     sslcon = myConnConfig(c);
 
-    sslcon->clienthello_vars = apr_palloc(c->pool, sizeof(*clienthello_vars));
+    sslcon->clienthello_vars = apr_pcalloc(c->pool, sizeof(*clienthello_vars));
     clienthello_vars = sslcon->clienthello_vars;
 
     clienthello_vars->version = SSL_client_hello_get0_legacy_version(ssl);
+
     clienthello_vars->ciphers_len = SSL_client_hello_get0_ciphers(ssl, &data);
-    clienthello_vars->ciphers_data = apr_pmemdup(c->pool, data, clienthello_vars->ciphers_len);
+    if (clienthello_vars->ciphers_len > 0) {
+        clienthello_vars->ciphers_data = apr_pmemdup(c->pool, data, clienthello_vars->ciphers_len);
+    } else {
+        clienthello_vars->ciphers_data = NULL;
```

Comment:
Since you're now using `apr_pcalloc` the explicit setting of NULL / 0 through here is not necessary.

Reply:
Done. I've removed the explicit/redundant assignments of _len to 0 and _data to NULL which should be fine as long as failed calls to the openssl client hello functions never modify _len. I've left the checks for _len >0 before memdup.

I've tested on conditions that triggered bugs due to uninitialized or 0 _len before by testing connections where some of the collected client hello extensions is not present and these cases functioned correctly.